modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/DTOOpenAPIParser.java

```diff
@@ -117,26 +118,13 @@ public static Schema getPropertySchema(String propertyName, Schema schema) {
 	public static boolean isSchemaProperty(
 		OpenAPIYAML openAPIYAML, String propertyName, Schema schema) {
 
-		Map<String, Schema> schemas = OpenAPIUtil.getAllSchemas(openAPIYAML);
-
 		Map<String, Schema> propertySchemas = _getPropertySchemas(schema);
 
 		for (Map.Entry<String, Schema> entry : propertySchemas.entrySet()) {
 			String propertySchemaName = entry.getKey();
-			Schema propertySchema = entry.getValue();
-
-			String curPropertyName = _getPropertyName(
-				propertySchema, propertySchemaName);
 
-			if (StringUtil.equalsIgnoreCase(curPropertyName, propertyName)) {
-				String schemaName = StringUtil.upperCaseFirstLetter(
-					propertySchemaName);
-
-				if (propertySchema.getItems() != null) {
-					schemaName = OpenAPIUtil.formatSingular(schemaName);
-				}
-
-				if (schemas.containsKey(schemaName)) {
+			if (propertySchemaName.equals(propertyName)) {
```

Comment:
I think this is giving false positives in the cases where there are properties with the same name than components but of other types than objects (For example the case of emailAddress on UserAccount)

Reply:
The old code did because it checked with schemas. 

But this code just finds the property object, it's comparing the name of the property with the schemas properties to get the full property object.